internal/core/build_info.go

```diff
@@ -14,9 +23,96 @@ type BuildInfo struct {
 	GoOS      string
 }
 
+const (
+	scwDisableCheckVersionEnv        = "SCW_DISABLE_CHECK_VERSION"
+	latestVersionFileURL             = "https://scw-devtools.s3.nl-ams.scw.cloud/scw-cli-v2-version"
+	latestVersionUpdateFileLocalName = "latest-cli-version"
+	latestVersionRequestTimeout      = 1 * time.Second
+)
+
 // IsRelease returns true when the version of the CLI is an official release:
 // - version must be non-empty (exclude tests)
-// - version must not contain label (e.g. '+dev')
+// - version must not contain metadata (e.g. '+dev')
 func (b *BuildInfo) IsRelease() bool {
-	return b.Version != "" && !strings.Contains(b.Version, "+")
+	return b.Version != nil && b.Version.Metadata() == ""
+}
+
+func (b *BuildInfo) checkVersion() {
+	if !b.IsRelease() || os.Getenv(scwDisableCheckVersionEnv) == "true" {
+		logger.Debugf("skipping check version")
+		return
+	}
+
+	latestVersionUpdateFilePath := filepath.Join(scw.GetCacheDirectory(), latestVersionUpdateFileLocalName)
+
+	// do nothing if last refresh at during the last 24h
+	if wasFileModifiedLast24h(latestVersionUpdateFilePath) {
+		logger.Debugf("version was already checked during past 24 hours")
+		return
+	}
+
+	// do nothing if we cannot create the file
+	if !createAndCloseFile(latestVersionUpdateFilePath) {
+		return
+	}
+
+	// pull latest version
+	latestVersion, err := getLatestVersion()
+	if err != nil {
+		logger.Debugf("failed to retrieve latest version: %s", err)
+		return
+	}
+
+	if b.Version.LessThan(latestVersion) {
+		logger.Infof("a new version of scw is available (%s), beware that you are currently running %v", b.Version, latestVersion)
```

Comment:
how is this shown to the user ? 

can you add a screenshoot please, to show the order of the command, and the warning. 

Is there a separation between the two ?

Reply:
Also, would it be possible to add a test ?

Reply:
done